frontend/views/containers/chatroom/CreateNewChannelModal.vue

```diff
@@ -83,22 +91,43 @@ export default ({
     ModalTemplate,
     BannerScoped
   },
+  computed: {
+    ...mapState(['currentGroupId'])
+  },
   data () {
     return {
       form: {
-        name: null,
-        description: null,
-        isPrivate: false
+        name: '',
+        description: '',
+        private: false
       }
     }
   },
   methods: {
     close () {
       this.$refs.modal.close()
     },
-    submit () {
-      console.log('TODO implement this.')
+    async submit () {
+      const { name, description } = this.form
+      try {
+        await sbp('gi.actions/group/addAndJoinChatRoom', {
+          contractID: this.currentGroupId,
+          data: {
+            attributes: {
+              name,
+              description,
+              privacyLevel: !this.form.private ? CHATROOM_PRIVACY_LEVEL.GROUP : CHATROOM_PRIVACY_LEVEL.PRIVATE,
+              type: CHATROOM_TYPES.GROUP
+            }
+          }
+        })
+      } catch (e) {
+        console.error('CreateNewChannelModal.vue submit() error:', e)
```

Comment:
Can you display this error using `v-error`?

Reply:
`this.$refs.formMsg.danger(L('Failed to create chat channel. {reportError}', LError(e)))`

Yeap. I think this is good solution. Thanks.